frontend/src/components/MapView/AlertForm/index.tsx

```diff
@@ -166,7 +166,7 @@ function AlertForm({ classes, isOpen, setOpen }: AlertFormProps) {
     };
 
     setAlertWaiting(true);
-    const response = await fetchApiData(ALERT_API_URL, request);
+    const response = await fetchApiData(ALERT_API_URL, request, dispatch);
```

Comment:
I found not friendly to pass the "dispatch" in all functions that need to call fetchWithTimeout function.

Do you have considered something like this?
> const response = await withTimeout(fetchApiData(ALERT_API_URL, request), dispatch, errorMessage);

In this cas withTimeout is a wrapper that throw an error when timeout is out and dispatch the error message. You don't have to modify fetchApiData function.

Reply:
I agree, is there anything we could do to make this a bit more dev-friendly?